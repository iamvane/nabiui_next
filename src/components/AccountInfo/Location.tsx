import * as React from 'react';
import PlacesAutocomplete from 'react-places-autocomplete';
import {
  TextField,
  Typography
} from '@material-ui/core';

import SectionTitle from 'components/common/SectionTitle';
import { LocationComponent } from 'components/AccountInfo/constants';

interface Props {
  handleLocationChange: (location: string) => void;
  handleLocationSelect: (location: string) => void;
  location: string;
}

const Location: React.StatelessComponent <Props> = props => {
  return (
    <div className="nabi-margin-top-large nabi-margin-bottom-xlarge">
      <SectionTitle text={LocationComponent.sectionTitle} />
      <Typography className="nabi-margin-top-xsmall">
        {LocationComponent.description}
      </Typography>
      <PlacesAutocomplete
        value={props.location}
        onChange={props.handleLocationChange}
        onSelect={props.handleLocationSelect}
      >
        {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
          <div>
            <TextField
              {...getInputProps({
                fullWidth: true
              })}
            />
            <div>
              {loading && <div>Loading...</div>}
              {suggestions.map(suggestion => {
                const className = suggestion.active
                  ? 'suggestion-item--active'
                  : 'suggestion-item';
                // inline style for demonstration purpose
                const style = suggestion.active
                  ? { backgroundColor: '#fafafa', cursor: 'pointer' }
                  : { backgroundColor: '#ffffff', cursor: 'pointer' };
                return (
                  <div
                    key={suggestion}
                    {...getSuggestionItemProps(suggestion, {
                      className,
                      style,
                    })}
                  >
                    <span>{suggestion.description}</span>
                  </div>
                );
              })}
            </div>
          </div>
        )}
      </PlacesAutocomplete>
    </div>
  );
};

export default Location;
