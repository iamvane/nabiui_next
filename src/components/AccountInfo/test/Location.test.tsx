import * as React from 'react';
import { shallow } from 'enzyme';
import Location from 'components/AccountInfo/Location';

describe('Location', () => {

  const wrapper = shallow(
    <Location
      handleLocationChange={jest.fn()}
      handleLocationSelect={jest.fn()}
      location="foo location"
    />
  );

  it('Renders the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
