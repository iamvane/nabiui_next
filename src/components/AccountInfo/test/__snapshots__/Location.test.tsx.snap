// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Location
    handleLocationChange={[MockFunction]}
    handleLocationSelect={[MockFunction]}
    location="foo location"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SectionTitle
          text="Location"
        />,
        <WithStyles(Typography)
          className="nabi-margin-top-xsmall"
        >
          Please enter your home address.
        </WithStyles(Typography)>,
        <PlacesAutocomplete
          debounce={200}
          highlightFirstSuggestion={false}
          onChange={[MockFunction]}
          onError={[Function]}
          onSelect={[MockFunction]}
          searchOptions={Object {}}
          shouldFetchSuggestions={true}
          value="foo location"
        >
          [Function]
        </PlacesAutocomplete>,
      ],
      "className": "nabi-margin-top-large nabi-margin-bottom-xlarge",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "text": "Location",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Please enter your home address.",
          "className": "nabi-margin-top-xsmall",
        },
        "ref": null,
        "rendered": "Please enter your home address.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": [Function],
          "debounce": 200,
          "highlightFirstSuggestion": false,
          "onChange": [MockFunction],
          "onError": [Function],
          "onSelect": [MockFunction],
          "searchOptions": Object {},
          "shouldFetchSuggestions": true,
          "value": "foo location",
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SectionTitle
            text="Location"
          />,
          <WithStyles(Typography)
            className="nabi-margin-top-xsmall"
          >
            Please enter your home address.
          </WithStyles(Typography)>,
          <PlacesAutocomplete
            debounce={200}
            highlightFirstSuggestion={false}
            onChange={[MockFunction]}
            onError={[Function]}
            onSelect={[MockFunction]}
            searchOptions={Object {}}
            shouldFetchSuggestions={true}
            value="foo location"
          >
            [Function]
          </PlacesAutocomplete>,
        ],
        "className": "nabi-margin-top-large nabi-margin-bottom-xlarge",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "text": "Location",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Please enter your home address.",
            "className": "nabi-margin-top-xsmall",
          },
          "ref": null,
          "rendered": "Please enter your home address.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": [Function],
            "debounce": 200,
            "highlightFirstSuggestion": false,
            "onChange": [MockFunction],
            "onError": [Function],
            "onSelect": [MockFunction],
            "searchOptions": Object {},
            "shouldFetchSuggestions": true,
            "value": "foo location",
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
